src/bitdrift_public/protobuf/workflow/v1/workflow.proto

```diff
@@ -161,6 +161,10 @@ message Workflow {
 
     // The match condition which is applied to the new value of the state during a state change.
     state.v1.StateValueMatch new_value = 4 [(validate.rules).message = {required: true}];
+
+    // Additional properties that must be true at the time of the state change for the rule to match.
+    // Since state changes do not match against a particular log, tags and messages will not be visibile to this matcher.
+    matcher.v1.LogMatcher log_matcher = 5;
```

Comment:
Agree this makes sense. It's kind of odd that we use the LogMatcher when we can only match on a part of it. Not sure if it's worth somehow factoring this out while it's still possible? If it's hard not a big deal.

Reply:
I could do another tree structure with new matcher lists etc but not sure if it's worth the complexity just for stronger typing, figured we could validate this during conversion and call it a day?